File: server/internal/service/knowledge.go
package service

import (
	"cy11dsphk/server/internal/database"
	"cy11dsphk/server/internal/model"
)

// ---------- PainPoint ----------

func ListPainPoints(merchantID uint64, keyword string) ([]model.PainPoint, error) {
	var list []model.PainPoint
	q := database.DB.Model(&model.PainPoint{})
	if merchantID > 0 {
		q = q.Where("merchant_id = ?", merchantID)
	}
	if keyword != "" {
		q = q.Where("content LIKE ? OR user_quote LIKE ? OR category LIKE ? OR suggested_topic LIKE ?",
			"%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%")
	}
	err := q.Order("id DESC").Find(&list).Error
	return list, err
}

func GetPainPoint(id uint64) (*model.PainPoint, error) {
	var m model.PainPoint
	if err := database.DB.First(&m, id).Error; err != nil {
		return nil, err
	}
	return &m, nil
}

func CreatePainPoint(m *model.PainPoint) error {
	return database.DB.Create(m).Error
}

func UpdatePainPoint(id uint64, m *model.PainPoint) error {
	m.ID = id
	return database.DB.Save(m).Error
}

func DeletePainPoint(id uint64) error {
	return database.DB.Delete(&model.PainPoint{}, id).Error
}

// ---------- CaseStudy ----------

func ListCaseStudies(merchantID uint64, keyword string) ([]model.CaseStudy, error) {
	var list []model.CaseStudy
	q := database.DB.Model(&model.CaseStudy{})
	if merchantID > 0 {
		q = q.Where("merchant_id = ?", merchantID)
	}
	if keyword != "" {
		q = q.Where("title LIKE ? OR account_name LIKE ? OR reusable_point LIKE ? OR industry LIKE ?",
			"%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%")
	}
	err := q.Order("id DESC").Find(&list).Error
	return list, err
}

func GetCaseStudy(id uint64) (*model.CaseStudy, error) {
	var m model.CaseStudy
	if err := database.DB.First(&m, id).Error; err != nil {
		return nil, err
	}
	return &m, nil
}

func CreateCaseStudy(m *model.CaseStudy) error {
	return database.DB.Create(m).Error
}

func UpdateCaseStudy(id uint64, m *model.CaseStudy) error {
	m.ID = id
	return database.DB.Save(m).Error
}

func DeleteCaseStudy(id uint64) error {
	return database.DB.Delete(&model.CaseStudy{}, id).Error
}

// ---------- AccountProfile ----------

func ListAccountProfiles(merchantID uint64, keyword string) ([]model.AccountProfile, error) {
	var list []model.AccountProfile
	q := database.DB.Model(&model.AccountProfile{})
	if merchantID > 0 {
		q = q.Where("merchant_id = ?", merchantID)
	}
	if keyword != "" {
		q = q.Where("account_name LIKE ? OR positioning LIKE ? OR industry LIKE ?",
			"%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%")
	}
	err := q.Order("id DESC").Find(&list).Error
	return list, err
}

func GetAccountProfile(id uint64) (*model.AccountProfile, error) {
	var m model.AccountProfile
	if err := database.DB.First(&m, id).Error; err != nil {
		return nil, err
	}
	return &m, nil
}

func CreateAccountProfile(m *model.AccountProfile) error {
	return database.DB.Create(m).Error
}

func UpdateAccountProfile(id uint64, m *model.AccountProfile) error {
	m.ID = id
	return database.DB.Save(m).Error
}

func DeleteAccountProfile(id uint64) error {
	return database.DB.Delete(&model.AccountProfile{}, id).Error
}

// ---------- PlatformRule ----------

func ListPlatformRules(merchantID uint64, keyword string) ([]model.PlatformRule, error) {
	var list []model.PlatformRule
	q := database.DB.Model(&model.PlatformRule{})
	if merchantID > 0 {
		q = q.Where("merchant_id = ?", merchantID)
	}
	if keyword != "" {
		q = q.Where("title LIKE ? OR content LIKE ? OR category LIKE ?",
			"%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%")
	}
	err := q.Order("id DESC").Find(&list).Error
	return list, err
}

func GetPlatformRule(id uint64) (*model.PlatformRule, error) {
	var m model.PlatformRule
	if err := database.DB.First(&m, id).Error; err != nil {
		return nil, err
	}
	return &m, nil
}

func CreatePlatformRule(m *model.PlatformRule) error {
	return database.DB.Create(m).Error
}

func UpdatePlatformRule(id uint64, m *model.PlatformRule) error {
	m.ID = id
	return database.DB.Save(m).Error
}

func DeletePlatformRule(id uint64) error {
	return database.DB.Delete(&model.PlatformRule{}, id).Error
}

// ---------- ContentTemplate ----------

func ListContentTemplates(merchantID uint64, keyword string) ([]model.ContentTemplate, error) {
	var list []model.ContentTemplate
	q := database.DB.Model(&model.ContentTemplate{})
	if merchantID > 0 {
		q = q.Where("merchant_id = ?", merchantID)
	}
	if keyword != "" {
		q = q.Where("name LIKE ? OR type LIKE ? OR category LIKE ? OR content LIKE ?",
			"%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%", "%"+keyword+"%")
	}
	err := q.Order("id DESC").Find(&list).Error
	return list, err
}

func GetContentTemplate(id uint64) (*model.ContentTemplate, error) {
	var m model.ContentTemplate
	if err := database.DB.First(&m, id).Error; err != nil {
		return nil, err
	}
	return &m, nil
}

func CreateContentTemplate(m *model.ContentTemplate) error {
	return database.DB.Create(m).Error
}

func UpdateContentTemplate(id uint64, m *model.ContentTemplate) error {
	m.ID = id
	return database.DB.Save(m).Error
}

func DeleteContentTemplate(id uint64) error {
	return database.DB.Delete(&model.ContentTemplate{}, id).Error
}

File: admin/apps/web-antd/src/api/admin/knowledge.ts
import { requestClient } from '#/api/request';

export interface KnowledgeListResult<T> {
  list: T[];
  total: number;
  page: number;
  size: number;
}

/* ===================== 痛点 pain-points ===================== */
export interface PainPoint {
  id: number;
  merchantId: number;
  merchantName: string;
  source: string;
  category: string;
  content: string;
  userQuote: string;
  emotion: string;
  product: string;
  demandLevel: string;
  suggestedTopic: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface PainPointPayload {
  merchantId?: number;
  source?: string;
  category?: string;
  content: string;
  userQuote?: string;
  emotion?: string;
  product?: string;
  demandLevel?: string;
  suggestedTopic?: string;
  status?: string;
}

export function getPainPoints(params: {
  keyword?: string;
  merchantId?: number;
  platform?: string;
  status?: string;
  page?: number;
  size?: number;
}) {
  return requestClient.get<KnowledgeListResult<PainPoint>>('/knowledge/pain-points', {
    params,
  });
}

export function getPainPoint(id: number) {
  return requestClient.get<PainPoint>(`/knowledge/pain-points/${id}`);
}

export function createPainPoint(data: Partial<PainPointPayload>) {
  return requestClient.post<PainPoint>('/knowledge/pain-points', data);
}

export function updatePainPoint(id: number, data: Partial<PainPointPayload>) {
  return requestClient.put<PainPoint>(`/knowledge/pain-points/${id}`, data);
}

export function deletePainPoint(id: number) {
  return requestClient.delete(`/knowledge/pain-points/${id}`);
}

/* ===================== 案例 case-studies ===================== */
export interface CaseStudy {
  id: number;
  merchantId: number;
  merchantName: string;
  title: string;
  platform: string;
  accountName: string;
  industry: string;
  form: string;
  hookType: string;
  structure: string;
  conversionAction: string;
  reusablePoint: string;
  dataJson: string;
  lesson: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface CaseStudyPayload {
  merchantId?: number;
  title: string;
  platform?: string;
  accountName?: string;
  industry?: string;
  form?: string;
  hookType?: string;
  structure?: string;
  conversionAction?: string;
  reusablePoint?: string;
  dataJson?: string;
  lesson?: string;
  status?: string;
}

export function getCaseStudies(params: {
  keyword?: string;
  merchantId?: number;
  platform?: string;
  status?: string;
  page?: number;
  size?: number;
}) {
  return requestClient.get<KnowledgeListResult<CaseStudy>>('/knowledge/case-studies', {
    params,
  });
}

export function getCaseStudy(id: number) {
  return requestClient.get<CaseStudy>(`/knowledge/case-studies/${id}`);
}

export function createCaseStudy(data: Partial<CaseStudyPayload>) {
  return requestClient.post<CaseStudy>('/knowledge/case-studies', data);
}

export function updateCaseStudy(id: number, data: Partial<CaseStudyPayload>) {
  return requestClient.put<CaseStudy>(`/knowledge/case-studies/${id}`, data);
}

export function deleteCaseStudy(id: number) {
  return requestClient.delete(`/knowledge/case-studies/${id}`);
}

/* ===================== 账号画像 account-profiles ===================== */
export interface AccountProfile {
  id: number;
  merchantId: number;
  merchantName: string;
  accountName: string;
  platform: string;
  industry: string;
  targetAudience: string;
  product: string;
  positioning: string;
  persona: string;
  toneStyle: string;
  monetization: string;
  updateFrequency: string;
  strengths: string;
  weaknesses: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface AccountProfilePayload {
  merchantId?: number;
  accountName: string;
  platform?: string;
  industry?: string;
  targetAudience?: string;
  product?: string;
  positioning?: string;
  persona?: string;
  toneStyle?: string;
  monetization?: string;
  updateFrequency?: string;
  strengths?: string;
  weaknesses?: string;
  status?: string;
}

export function getAccountProfiles(params: {
  keyword?: string;
  merchantId?: number;
  platform?: string;
  status?: string;
  page?: number;
  size?: number;
}) {
  return requestClient.get<KnowledgeListResult<AccountProfile>>(
    '/knowledge/account-profiles',
    { params },
  );
}

export function getAccountProfile(id: number) {
  return requestClient.get<AccountProfile>(`/knowledge/account-profiles/${id}`);
}

export function createAccountProfile(data: Partial<AccountProfilePayload>) {
  return requestClient.post<AccountProfile>('/knowledge/account-profiles', data);
}

export function updateAccountProfile(id: number, data: Partial<AccountProfilePayload>) {
  return requestClient.put<AccountProfile>(`/knowledge/account-profiles/${id}`, data);
}

export function deleteAccountProfile(id: number) {
  return requestClient.delete(`/knowledge/account-profiles/${id}`);
}

/* ===================== 平台规则 platform-rules ===================== */
export interface PlatformRule {
  id: number;
  merchantId: number;
  merchantName: string;
  platform: string;
  category: string;
  title: string;
  content: string;
  riskLevel: string;
  source: string;
  effectiveDate: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface PlatformRulePayload {
  merchantId?: number;
  platform: string;
  category?: string;
  title: string;
  content?: string;
  riskLevel?: string;
  source?: string;
  effectiveDate?: string;
  status?: string;
}

export function getPlatformRules(params: {
  keyword?: string;
  platform?: string;
  status?: string;
  page?: number;
  size?: number;
}) {
  return requestClient.get<KnowledgeListResult<PlatformRule>>('/knowledge/platform-rules', {
    params,
  });
}

export function getPlatformRule(id: number) {
  return requestClient.get<PlatformRule>(`/knowledge/platform-rules/${id}`);
}

export function createPlatformRule(data: Partial<PlatformRulePayload>) {
  return requestClient.post<PlatformRule>('/knowledge/platform-rules', data);
}

export function updatePlatformRule(id: number, data: Partial<PlatformRulePayload>) {
  return requestClient.put<PlatformRule>(`/knowledge/platform-rules/${id}`, data);
}

export function deletePlatformRule(id: number) {
  return requestClient.delete(`/knowledge/platform-rules/${id}`);
}

/* ===================== 内容模板 content-templates ===================== */
export interface ContentTemplate {
  id: number;
  merchantId: number;
  merchantName: string;
  name: string;
  type: string;
  category: string;
  structureJson: string;
  content: string;
  usageNote: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface ContentTemplatePayload {
  merchantId?: number;
  name: string;
  type?: string;
  category?: string;
  structureJson?: string;
  content?: string;
  usageNote?: string;
  status?: string;
}

export function getContentTemplates(params: {
  keyword?: string;
  merchantId?: number;
  platform?: string;
  status?: string;
  page?: number;
  size?: number;
}) {
  return requestClient.get<KnowledgeListResult<ContentTemplate>>(
    '/knowledge/content-templates',
    { params },
  );
}

export function getContentTemplate(id: number) {
  return requestClient.get<ContentTemplate>(`/knowledge/content-templates/${id}`);
}

export function createContentTemplate(data: Partial<ContentTemplatePayload>) {
  return requestClient.post<ContentTemplate>('/knowledge/content-templates', data);
}

export function updateContentTemplate(id: number, data: Partial<ContentTemplatePayload>) {
  return requestClient.put<ContentTemplate>(`/knowledge/content-templates/${id}`, data);
}

export function deleteContentTemplate(id: number) {
  return requestClient.delete(`/knowledge/content-templates/${id}`);
}
